llvm/lib/Transforms/IPO/FunctionSpecialization.cpp

```diff
@@ -889,18 +910,20 @@ bool FunctionSpecializer::findSpecializations(Function *F, unsigned FuncSize,
       unsigned Score = 0;
       InstCostVisitor Visitor = getInstCostVisitorFor(F);
       for (ArgInfo &A : S.Args) {
-        CodeSize += Visitor.getCodeSizeBonus(A.Formal, A.Actual);
+        CodeSize += Visitor.getCodeSizeSavingsForArg(A.Formal, A.Actual);
         Score += getInliningBonus(A.Formal, A.Actual);
       }
-      CodeSize += Visitor.getCodeSizeBonusFromPendingPHIs();
+      CodeSize += Visitor.getCodeSizeSavingsFromPendingPHIs();
 
       LLVM_DEBUG(dbgs() << "FnSpecialization: Specialization bonus {CodeSize = "
                         << CodeSize << ", Inlining = " << Score << "}\n");
 
-      Bonus B = {CodeSize, 0};
-      FunctionGrowth[F] += FuncSize - B.CodeSize;
+      unsigned LatencySavings = 0;
+      unsigned CodeSizeSavings = getCostValue(CodeSize);
+      FunctionGrowth[F] += FuncSize - CodeSizeSavings;
 
```

Comment:
Can we also sink all three of the above statements inside the lambda, to the point of their use? Since this isn't a read-only lambda anymore I think it makes sense to get rid of the parameters and capture the scope by reference.

Reply:
I've sunk/moved things into the lambda and refactored it to just capture the scope by reference.

Note - I've realized that we accumulate the CodeSize increase into FunctionGrowth[F] for all specializations that we evaluate, rather than just the actual specializations we create. I've added a TODO for fixing this, as I think the changes would be non-trivial and don't make sense to do in this PR.